pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -409,18 +414,13 @@ async function resolveDependenciesOfImporters (
       const postponedPeersResolutionQueue: PostponedPeersResolutionFunction[] = []
       const pkgAddresses: PkgAddress[] = []
 
-      const resolvedDependenciesOfImporter = await Promise.all(
-        extendedWantedDeps.map((extendedWantedDep) => resolveDependenciesOfDependency(
+      const resolvedDependenciesOfImporter = await Promise.all(extendedWantedDeps.map(extendedWantedDep =>
+        resolveDependenciesOfImporterDependency({
           ctx,
-          importer.preferredVersions,
-          {
-            ...importer.options,
-            parentPkgAliases: importer.parentPkgAliases,
-            pickLowestVersion: pickLowestVersion && !importer.updatePackageManifest,
-          },
-          extendedWantedDep
-        ))
-      )
+          extendedWantedDep,
+          importer,
+          pickLowestVersion,
+        })))
```

Comment:
if you put extendedWantedDep as the last argument, you can bind all the other options and only provide the extendedWantedDep when called inside the map

Reply:
Done!

Reply:
I'm a bit more used to using lambdas, but will make a note to try and use `.bind` more when I work on pnpm.

I think one thing that's a bit unfortunate is the need to re-arrange parameters for `.bind` since you can't do that for external functions in NPM libraries.